src/pluginLoader.js

```diff
@@ -0,0 +1,660 @@
+ *
+ * TODO Figure out a better strategy for API discovery so that this can be
+ * removed altogether.
+ *
+ */
+
+// log initialization
+/* globals Logger window console document localStorage $ angular jQuery navigator Jolokia */
+(function() {
+  'use strict';
+
+  Logger.setLevel(Logger.INFO);
+  Logger.storagePrefix = 'hawtio';
+
+  Logger.oldGet = Logger.get;
+  Logger.loggers = {};
+  Logger.get = function(name) {
+    var answer = Logger.oldGet(name);
+    Logger.loggers[name] = answer;
+    return answer;
+  };
+
+  // we'll default to 100 statements I guess...
+  window['LogBuffer'] = 100;
+
+  if ('localStorage' in window) {
+    if (!('logLevel' in window.localStorage)) {
+      window.localStorage['logLevel'] = JSON.stringify(Logger.INFO);
+    }
+    var logLevel = Logger.DEBUG;
+    try {
+      logLevel = JSON.parse(window.localStorage['logLevel']);
+    } catch (e) {
+      console.error("Failed to parse log level setting: ", e);
+    }
+    // console.log("Using log level: ", logLevel);
+    Logger.setLevel(logLevel);
+    if ('showLog' in window.localStorage) {
+      var showLog = window.localStorage['showLog'];
+      // console.log("showLog: ", showLog);
+      if (showLog === 'true') {
+        var container = document.getElementById("log-panel");
+        if (container) {
+          container.setAttribute("style", "bottom: 50%;");
+        }
+      }
+    }
+    if ('logBuffer' in window.localStorage) {
+      var logBuffer = window.localStorage['logBuffer'];
+      window['LogBuffer'] = parseInt(logBuffer, 10);
+    } else {
+      window.localStorage['logBuffer'] = window['LogBuffer'];
+    }
+    if ('childLoggers' in window.localStorage) {
+      var childLoggers = [];
+      try {
+        childLoggers = JSON.parse(localStorage['childLoggers']);
+      } catch (e) {
+
+      }
+      childLoggers.forEach(function(child) {
+        Logger.get(child.logger).setLevel(Logger[child.level]);
+      });
+    }
+  }
+
+  var consoleLogger = null;
+
+  if ('console' in window) {
+    window['JSConsole'] = window.console;
+    consoleLogger = function(messages, context) {
+      var MyConsole = window['JSConsole'];
+      var hdlr = MyConsole.log;
+      // Prepend the logger's name to the log message for easy identification.
+      if (context.name) {
+        messages[0] = "[" + context.name + "] " + messages[0];
+      }
+      // Delegate through to custom warn/error loggers if present on the console.
+      if (context.level === Logger.WARN && 'warn' in MyConsole) {
+        hdlr = MyConsole.warn;
+      } else if (context.level === Logger.ERROR && 'error' in MyConsole) {
+        hdlr = MyConsole.error;
+      } else if (context.level === Logger.INFO && 'info' in MyConsole) {
+        hdlr = MyConsole.info;
+      }
+      if (hdlr && hdlr.apply) {
+        try {
+          hdlr.apply(MyConsole, messages);
+        } catch (e) {
+          MyConsole.log(messages);
+        }
+      }
+    };
+  }
```

Comment:
Let's delete lines 70-97 and use js-logger default handler to add the context name. (I know this is existing hawtio code.)

https://github.com/jonnyreeves/js-logger/blob/46f9df7b946d50574601bb394e5d39c9ebcea338/README.md#default-log-handler-function